tgui/packages/tgui/interfaces/MODsuit.tsx

```diff
@@ -0,0 +1,723 @@
+                tooltipPosition="top"
+              />
+            </Table.Cell>
+          </Table.Row>
+          {module_info.map((module) => {
+            return (
+              <Table.Row key={module.ref}>
+                <Table.Cell width={1}>
+                  <Button
+                    onClick={() => act('select', { 'ref': module.ref })}
+                    icon={
+                      module.module_type === 3
+                        ? module.module_active
+                          ? 'square-check'
+                          : 'square'
+                        : 'power-off'
+                    }
+                    selected={module.module_active}
+                    tooltip={moduleTypeAction(module.module_type)}
+                    tooltipPosition="left"
+                    disabled={!module.module_type}
+                  />
+                </Table.Cell>
+                <Table.Cell width={1}>
+                  <Button
+                    onClick={() => setConfigureState(module.ref)}
+                    icon="cog"
+                    selected={configureState === module.ref}
+                    tooltip="Configure"
+                    tooltipPosition="left"
+                    disabled={module.configuration_data.length === 0}
+                  />
+                </Table.Cell>
+                <Table.Cell width={1}>
+                  <Button
+                    onClick={() => act('pin', { 'ref': module.ref })}
+                    icon="thumbtack"
+                    selected={module.pinned}
+                    tooltip="Pin"
+                    tooltipPosition="left"
+                    disabled={!module.module_type}
+                  />
+                </Table.Cell>
+                <Table.Cell>
+                  <Collapsible
+                    title={module.module_name}
+                    color={module.module_active ? 'green' : 'default'}>
+                    <Section mr={-19}>{module.description}</Section>
+                  </Collapsible>
+                  {configureState === module.ref && (
+                    <ConfigureScreen
+                      configuration_data={module.configuration_data}
+                      module_ref={module.ref}
+                      module_name={module.module_name}
+                      onExit={() => setConfigureState('')}
+                    />
+                  )}
+                </Table.Cell>
+                <Table.Cell textAlign="center">{module.idle_power}</Table.Cell>
+                <Table.Cell textAlign="center">
+                  {module.active_power}
+                </Table.Cell>
+                <Table.Cell textAlign="center">{module.use_power}</Table.Cell>
+                <Table.Cell textAlign="center">
+                  {module.module_complexity}
+                </Table.Cell>
+              </Table.Row>
+            );
+          })}
+        </Table>
+      )) || <NoticeBox>No Modules Detected</NoticeBox>}
```

Comment:
Not terrible but I feel like this makes it easy to lose your place. Can you early return this?